docs/specs/delete-user.md

```diff
@@ -0,0 +1,22 @@
+# Delete User
+
+  * [Deleting user](#deleting-user)
+  * [Cached data](#cached-data)
+  * [Future works](#future-works)
+
+## Deleting user
+
+A user can be deleted by admins. A deleted user would have all its associated
+data erased from database (hard-delete), including identities, authenticators,
+password data, etc. 
+
+## Cached data
+
+Some data may be present in caches (Redis), such as sessions, rate limit
```

Comment:
We should also remove sessions as part of the deletion process, as we treat sessions as entity as well.